src/server/dstu2/observation/controllers/observation.controller.js

```diff
@@ -41,10 +39,24 @@ module.exports.getObservations = (profile, logger, config) => {
 							'fullUrl': `${config.auth.resourceServer}/dstu2/Observation/${resource.id}`
 						};
 						searchResults.entry.push(entry);
+						// if req.patient is present, they are only allowed to access observations
+						// which reference a patient with their id, so if any resource.id
+						// does not equal the req.patient, then they are attempting to access
+						// a patients records that they are not allowed to access
+						if (req.patient && req.patient !== entry.resource.patientId) {
```

Comment:
Is this really what should happen here?  It means that a poorly-implemented backend service that doesn't do a patient check and just returns the search results would result in a patient never seeing any records unless they craft a query that returns exactly their result and no others.  In fact, that makes it possible for a user to deduce which attributes their account does not share with any other accounts, which technically violates PHI protections.

Instead, we should simply filter out any results that aren't the patient.  That way we always return either 1 or 0 results but you can't tell whether they are any other records that match your search.

Reply:
I'll try to dig more into the documentation to find out what we do in case of scope errors.

Launch context parameter
"patient"

Example value
"123"

Meaning
String value with a patient id, indicating that the app was launched in the context of FHIR Patient 123. If the app has any patient-level scopes, they will be scoped to Patient 123.

Reply:
We can change this to do that, but I am not sure that is what we are supposed to do. I will need to try to find where it talks about handling these situations in the spec.  My original understanding was that if a patient is authorized, and has the scopes to access this endpoint, you should return a 403 whenever they attempt to access records they are not allowed to access. Ill see if I can find something in the docs for this situation.